### PR TITLE
refactor the stream seek function signature so it matches the stream read function.

### PR DESCRIPTION
**Pull request by stevenhoving**

### CODE REVIEW

File: src/streams/tests/test_libaeon_streams/test_stream_reader.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@
 struct test_fixture_stream_reader_default_data : public ::testing::Test
 {
     test_fixture_stream_reader_default_data()
-        : fixture_data({{'A', 'B', 'C', 'D', 'E', '\n', '1', '2', '3', '4'}})
+        : fixture_data({{'A', 'B', 'C', 'D', 'E', '\r', '\n', '1', '2', '3', '4'}})
```
Comment (robindegen) [inline src/streams/tests/test_libaeon_streams/test_stream_reader.cpp:38]: regular \n should also work though? It has to support both

File: src/streams/include/aeon/streams/stream_reader.h [hunk 2]
```diff
@@ -49,6 +49,7 @@ namespace streams
 template <typename T>
 class stream_reader : common::noncopyable
 {
+    static constexpr int read_line_block_size = 64;
```
Comment (stevenhoving) [inline src/streams/include/aeon/streams/stream_reader.h:53]: i'm still wondering if 64 bytes is to low...

File: src/streams/tests/test_libaeon_streams/test_stream_reader.cpp (hunk 1 above)
Comment (stevenhoving) [reply src/streams/tests/test_libaeon_streams/test_stream_reader.cpp:38]: The point was, that '\r' wasn't tested. By adding '\r' to this test string, both '\r' and '\n' behavior is tested.

Comment (robindegen) [reply src/streams/tests/test_libaeon_streams/test_stream_reader.cpp:38]: Makes sense 👍

File: src/streams/include/aeon/streams/stream_reader.h (hunk 2 above)
Comment (robindegen) [reply src/streams/include/aeon/streams/stream_reader.h:53]: Can always make it a cmake configurable thing later

**Comment by stevenhoving** [comment]
squashed commits that failed to build.

**Pull request merged**